src/ast-utils.ts

```diff
@@ -103,6 +103,39 @@ export function removeElementAndNewline(node: ASTNode, replacement?: ASTNode) {
   }
 }
 
+const injectedTagNames = new Set(['html', 'head', 'body']);
+/**
+ * When parse5 parses an HTML document, it tries to fill in a few html tags
+ * it considers missing if it doesn't see them (see `injectedTagNames` const
+ * above.)  This function removes these elements from the AST so the AST
+ * represents only the html that was parsed.  The primary signal is that the
+ * node has no `__location` information, so this function can only reliably
+ * be used on a fresh parse, since subsequent tree manipulations may inject
+ * nodes without location information.
+ */
+export function removeFakeNodes(ast: dom5.Node) {
```

Comment:
Please add a TODO to use the common version of this once it lands in either dom5 or the analyzer